src/app/participant-page/participant-page.component.ts

```diff
@@ -1100,25 +1101,37 @@ export class ParticipantPageComponent implements OnInit, OnDestroy {
     );
   }
 
-  getParticipantData(fieldSetting: FieldSettings, relative: ParticipantData) {
-    if (this.participant != null && this.participant.participantData != null && relative.dataId != null && fieldSetting.columnName != null) {
-      let kitFieldsDict = {'DATE_KIT_RECEIVED': 'receiveDate', 'DATE_KIT_SENT': 'scanDate', 'KIT_TYPE_TO_REQUEST': 'kitType'};
-      if (fieldSetting.displayType === 'SAMPLE') {
-        let sample: Sample = this.participant.kits.find(kit => kit.bspCollaboratorSampleId === relative.data[fieldSetting.columnName]);
-        if (sample && kitFieldsDict[fieldSetting.columnName] && sample[kitFieldsDict[fieldSetting.columnName]]) {
-          return sample[kitFieldsDict[fieldSetting.columnName]];
-        } else {
-          return "";
+  getParticipantData(fieldSetting: FieldSettings, personsParticipantData: ParticipantData) {
+    if (this.participant && this.participant.participantData && personsParticipantData && fieldSetting.columnName) {
+      if (personsParticipantData && personsParticipantData.data) {
+        if (personsParticipantData.data[fieldSetting.columnName]) {
+          return personsParticipantData.data[fieldSetting.columnName];
+        } else if (fieldSetting.actions && fieldSetting.actions[0]) {
+          if (fieldSetting.actions[0].type === 'CALC' && fieldSetting.actions[0].value && personsParticipantData.data[fieldSetting.actions[0].value]) {
+            return this.countYears(personsParticipantData.data[fieldSetting.actions[0].value]);
+          } else if (fieldSetting.actions[0].type === 'SAMPLE' && fieldSetting.actions[0].type2 === 'MAP_TO_KIT') {
+            return this.getSampleFieldValue(fieldSetting, personsParticipantData);
+          }
         }
       }
-      let participantData = this.participant.participantData.find(participantData => participantData.dataId === relative.dataId);
+    }
+    return "";
+  }
+
+  getParticipantDataForAT(fieldSetting: FieldSettings, fieldTypeId: string) {
```

Comment:
I don't like having the study name here in the method, makes it feel like it would only work for AT and not being dynamic

Reply:
Yes, I agree, changed it